src/tally.rs

```diff
@@ -33,61 +58,73 @@ async fn days() -> Result<(Vec<(PathBuf, u32)>, Vec<u32>), AocError>
         }
     }
 
-
     Ok((have, dont_have))
 }
 
-async fn build_day(day: u32) -> Result<(), AocError>
+async fn build_days(cargo_folder: PathBuf, days: &[(PathBuf, u32)]) -> Result<Vec<u32>, AocError>
 {
-    let bin = format!("day_{:02}", day);
-    let res = tokio::process::Command::new("cargo")
-        .args(["build", "--release", "--bin", &bin])
-        .output()
-        .await?;
+    let sty = ProgressStyle::with_template(
+        "[{elapsed_precise}] {msg}... {bar:40.cyan/blue} {pos:>7}/{len:7}",
+    )
+    .unwrap()
+    .progress_chars("##-");
 
-    if !res.status.success()
-    {
-        Err(AocError::BuildError(bin.to_string()))
-    }
-    else
-    {
-        Ok(())
-    }
-}
+    let progress = ProgressBar::new(days.len() as u64);
+    progress.set_style(sty);
+    progress.set_message("compiling");
 
-async fn build_days(cargo_folder: PathBuf, days: &[(PathBuf, u32)]) -> Result<Vec<u32>, AocError>
-{
-    let mut set = JoinSet::new();
-    for (_path, day) in days.iter()
-    {
+    let res: Result<(), AocError> = thread_exec(days, |(_path, day)| {
         let day = *day;
-        set.spawn(async move { build_day(day).await });
-    }
+        let progress = progress.clone();
+        let bin = format!("day_{:02}", day);
+        let res = std::process::Command::new("cargo")
+            .args(["build", "--release", "--bin", &bin])
+            .output()?;
 
-    while let Some(Ok(res)) = set.join_next().await
-    {
-        res?;
-    }
+        progress.inc(1);
+        if !res.status.success()
+        {
+            Err(AocError::BuildError(bin))
+        }
+        else
+        {
+            Ok(())
+        }
+    });
+    res?;
 
-    let mut vec = Vec::new();
-    for (pb, day) in days
-    {
+    progress.reset();
+    progress.set_message("verifying days");
+
+    let unimpls: Vec<Option<u32>> = thread_exec(days, |(pb, day)| {
         let pb = pb.clone();
         let day = *day;
-        if run_day(cargo_folder.clone(), (pb, day), 1).await.is_err()
-        {
-            vec.push(day);
-        }
-    }
+        let bin = format!("day_{:02}", day);
+        let mut target = cargo_folder.clone();
+        target.push("target/release");
+        target.push(&bin);
+        let progress = progress.clone();
 
-    Ok(vec)
+        let res = match std::process::Command::new(target).current_dir(&pb).output()
+        {
+            Ok(res) =>
+            {
+                // collect all 'unimplemented' days
+                parse_get_times(res).map_err(|_| day).err()
+            },
+            Err(_) => Some(day),
+        };
+        progress.inc(1);
+        res
+    });
+    Ok(unimpls.into_iter().flatten().collect())
 }
 
 fn parse_get_times(output: Output) -> Result<(usize, Option<usize>), AocError>
```

Comment:
I think we need to be smarter about this (but that can definitely be for another PR IMO). Right now, it considers a day that prints something other than the actual answer as a day that is not implemented. It's not a big problem though as I think when cleaning up and "finishing" up a day, one often removes the other prints 🤷

Reply:
I'll create an issue for this, and then we can resolve this conversation